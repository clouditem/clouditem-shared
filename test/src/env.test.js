require('./../finally');
const sut = require('./../../src/env.js');

/* eslint-disable padded-blocks,brace-style*/
test.each`
input                            | env               | requested          |  test_name
${{PORT: undefined}}             |  ${{PORT: 8080}}  | ${['PORT']}        | ${'property present'}
${{PORT: undefined}}             |  ${{}}            | ${['PORT']}        | ${'property required but not present'}
${{PORT: 8080}}                  |  ${{}}            | ${['PORT']}        | ${'property absent, default set'}
${{PORT: port => port + 1}}      |  ${{PORT: 8080}}  | ${['PORT']}        | ${'func increment by one'}
${{}}                            |  ${{}}            | ${['PORT']}        | ${'property absent, but requested'}
${{PORT: () => { throw Error('test'); }}}   |  ${{PORT: 8080}}  | ${['PORT']}        | ${'func throw error with message test'}
`('$test_name : $env', ({input, env, requested}) => {
/* eslint-enable padded-blocks,brace-style*/
  process.env = env;
  let promise = Promise.resolve(Error('Error in Test'));
  try {

    const environment = sut(input);
    promise = Promise.all(requested.map(name => Promise.resolve()
      .then(() => ({[name]: environment[name]}))
    ))
      .then(properties => properties.reduce((accumulator, property) => Object.assign(accumulator, property), {}))
      .catch(err => err);

  } catch (err) {

    promise = Promise.resolve(err);

  }
  return promise.then(result => expect(result).toMatchSnapshot());

}
);
