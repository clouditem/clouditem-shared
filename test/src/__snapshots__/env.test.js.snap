// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"func increment by one" : {"PORT": 8080} 1`] = `
Object {
  "PORT": 8081,
}
`;

exports[`"func throw error with message test" : {"PORT": 8080} 1`] = `[Error: test]`;

exports[`"property absent, but requested" : {} 1`] = `[ReferenceError: zealit: property 'PORT' is nonexistent]`;

exports[`"property absent, default set" : {} 1`] = `
Object {
  "PORT": 8080,
}
`;

exports[`"property present" : {"PORT": 8080} 1`] = `
Object {
  "PORT": 8080,
}
`;

exports[`"property required but not present" : {} 1`] = `[Error: environment property PORT was not found and has no default]`;
